src/context.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import * as Emitter from 'tiny-emitter';
+
+interface PortalProviderProps {}
+
+export interface PortalValue {
+  subscribeToPortal: Function;
+  unsubscribeFromPortal: Function;
+  setPortal: Function;
+  getPortal: Function;
+}
+
+const PortalContext = React.createContext<PortalValue | null>(null);
+
+export class PortalProvider extends React.PureComponent<PortalProviderProps> {
+  private emitter: Emitter.TinyEmitter | null = null;
+  private portals: Map<string, any[]> = new Map();
```

Comment:
Can we be more specific about than `any[]`? Does `React.ReactNode` cover all the use cases?

Reply:
a83f655